File: gracie.lee/src/main/java/homework/chapter_6/bonus2/EventOrder.java
package homework.chapter_6.bonus2;

import jakarta.persistence.*;
import java.time.LocalDateTime;

/**
 * 이벤트 주문 엔티티 (복합 키 사용)
 *
 * PK = (member_id, product_id)
 * → 같은 회원이 같은 상품을 2번 주문 불가 (1인 1회 제한)
 */
@Entity(name = "Bonus2EventOrder")
@Table(name = "event_order_bonus2")
@IdClass(EventOrderId.class)
public class EventOrder {

    @Id
    @ManyToOne
    @JoinColumn(name = "member_id")
    private Member member;

    @Id
    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    private LocalDateTime orderDate;

    protected EventOrder() {
    }

    public Member getMember() {
        return member;
    }

    public Product getProduct() {
        return product;
    }

    public LocalDateTime getOrderDate() {
        return orderDate;
    }

    public void setMember(Member member) {
        this.member = member;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public void setOrderDate(LocalDateTime orderDate) {
        this.orderDate = orderDate;
    }
}

File: joon.bug/src/main/kotlin/com/example/joonbug/chapter6/bonus2/EventOrder.kt
package com.example.joonbug.chapter6.bonus2

import com.example.joonbug.chapter6.homework3.Member
import com.example.joonbug.chapter6.homework3.Product
import jakarta.persistence.*
import java.time.LocalDate

/**
 * 보너스 2: 복합 키로 1인 1회 구매 제약 구현
 *
 * (회원, 상품)을 복합 키로 사용하여 같은 회원이 같은 상품을
 * 두 번 구매할 수 없도록 비즈니스 제약 구현
 */
@Entity(name = "HwBonus2EventOrder")
@Table(name = "hw_bonus2_event_order")
@IdClass(EventOrderId::class)
class EventOrder(
    @Id
    @ManyToOne
    @JoinColumn(name = "member_id")
    var member: Member? = null,

    @Id
    @ManyToOne
    @JoinColumn(name = "product_id")
    var product: Product? = null,

    var count: Int = 0,

    var orderDate: LocalDate? = null
)